packages/server-wallet/src/__test__/create-and-close-channel/ledger-funding.test.ts

```diff
@@ -0,0 +1,590 @@
+  await a.dbAdmin().createDB();
+  await b.dbAdmin().createDB();
+  await Promise.all([a.dbAdmin().migrateDB(), b.dbAdmin().migrateDB()]);
+
+  participantA = {
+    signingAddress: await a.getSigningAddress(),
+    participantId: 'a',
+    destination: makeDestination(
+      '0xaaaa000000000000000000000000000000000000000000000000000000000001'
+    ),
+  };
+  participantB = {
+    signingAddress: await b.getSigningAddress(),
+    participantId: 'b',
+    destination: makeDestination(
+      '0xbbbb000000000000000000000000000000000000000000000000000000000002'
+    ),
+  };
+});
+
+afterAll(async () => {
+  await Promise.all([a.destroy(), b.destroy()]);
+  await a.dbAdmin().dropDB();
+  await b.dbAdmin().dropDB();
+});
+
+/**
+ * Create a directly funded channel that will be used as the ledger channel.
+ *
+ * Note that this is just a simplification of the direct-funding test.
+ */
+const createLedgerChannel = async (): Promise<string> => {
+  const aDepositAmtETH = BN.from(10);
+  const bDepositAmtETH = BN.from(10);
+  const ledgerChannelArgs = {
+    participants: [participantA, participantB],
+    allocations: [
+      {
+        allocationItems: [
+          {
+            destination: participantA.destination,
+            amount: aDepositAmtETH,
+          },
+          {
+            destination: participantB.destination,
+            amount: bDepositAmtETH,
+          },
+        ],
+        token: ETH_ASSET_HOLDER_ADDRESS,
+      },
+    ],
+    appDefinition: ethers.constants.AddressZero,
+    appData: '0x00',
+    fundingStrategy: 'Direct' as const,
+  };
+  const resultA0 = await a.createChannel(ledgerChannelArgs);
+  const channelId = resultA0.channelResults[0].channelId;
+  await b.pushMessage(getPayloadFor(participantB.participantId, resultA0.outbox));
+  const resultB1 = await b.joinChannel({channelId});
+  await a.pushMessage(getPayloadFor(participantA.participantId, resultB1.outbox));
+  const fundingPostADeposit = {
+    channelId,
+    token: ETH_ASSET_HOLDER_ADDRESS,
+    amount: aDepositAmtETH,
+  };
+  await a.updateFundingForChannels([fundingPostADeposit]);
+  await b.updateFundingForChannels([fundingPostADeposit]);
+  const fundingPostBDeposit = {
+    channelId,
+    token: ETH_ASSET_HOLDER_ADDRESS,
+    amount: BN.add(aDepositAmtETH, bDepositAmtETH),
+  };
+  const resultA2 = await a.updateFundingForChannels([fundingPostBDeposit]);
+  await b.updateFundingForChannels([fundingPostBDeposit]);
+  const resultB3 = await b.pushMessage(getPayloadFor(participantB.participantId, resultA2.outbox));
+  await a.pushMessage(getPayloadFor(participantA.participantId, resultB3.outbox));
+
+  a.__setLedger(channelId, ETH_ASSET_HOLDER_ADDRESS);
+  b.__setLedger(channelId, ETH_ASSET_HOLDER_ADDRESS);
+
+  return channelId;
+};
+
+afterEach(() => {
+  a.store.ledgers = {};
+  b.store.ledgers = {};
+});
+
+describe('Funding a single channel', () => {
+  it('can fund a channel by ledger between two wallets ', async () => {
```

Comment:
To make sure I understand, this test is asserting the specific sequence of messages passed between the two wallets, in this specific scenario, with a particular ordering of messages?